pkg/referrerstore/oras/authprovider/azure/azureidentity.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+IdentityTHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	provider "github.com/deislabs/ratify/pkg/referrerstore/oras/authprovider"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/services/preview/containerregistry/runtime/2019-08-15-preview/containerregistry"
+)
+
+type azureManagedIdentityProviderFactory struct{}
+type azureManagedIdentityAuthProvider struct {
+	identityToken azcore.AccessToken
+	clientID      string
+	tenantID      string
+}
+
+type azureManagedIdentityAuthProviderConf struct {
+	Name     string `json:"name"`
+	ClientID string `json:"clientID"`
+}
+
+var (
+	// ErrorNoAuth indicates that no credentials are provided.
+	ErrorNoAuth    = fmt.Errorf("no credentials provided for Azure cloud provider")
+	ErrorNilEnv    = fmt.Errorf("env is nil pointer for GetServicePrincipalToken")
+	ErrorNilConfig = fmt.Errorf("config is nil pointer for GetServicePrincipalToken")
+)
+
+const (
+	azureManagedIdentityAuthProviderName string = "azureManagedIdentity"
+)
+
+// init calls Register for our Azure Workload Identity provider
+func init() {
+	provider.Register(azureManagedIdentityAuthProviderName, &azureManagedIdentityProviderFactory{})
+}
+
+// Create returns an azureManagedIdentityAuthProvider
+func (s *azureManagedIdentityProviderFactory) Create(authProviderConfig provider.AuthProviderConfig) (provider.AuthProvider, error) {
+	conf := azureManagedIdentityAuthProviderConf{}
+	authProviderConfigBytes, err := json.Marshal(authProviderConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(authProviderConfigBytes, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse auth provider configuration: %v", err)
```

Comment:
should this "failed to parse" message be more helpful to include name of the provider?  "fail to parse AzureManagedIdentity configuration" maybe?

Reply:
added